scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/SparkBackend.scala

```diff
@@ -11,6 +11,38 @@ import scala.collection.mutable.{ Map => MMap, ArrayBuffer }
 
 object SparkPlanner {
   import SparkMode.SparkConfigMethods
+  sealed trait PartitionComputer {
+    def apply(currentNumPartitions: Int): Int
+  }
+
+  final case object IdentityPartitionComputer extends PartitionComputer {
+    def apply(currentNumPartitions: Int): Int = currentNumPartitions
+  }
+  final case class ConfigPartitionComputer(config: Config, scaldingReducers: Option[Int]) extends PartitionComputer {
+    def apply(currentNumPartitions: Int): Int = {
+      val maxPartitions = config.getMaxPartitionCount
+      val getReducerScaling = config.getReducerScaling.getOrElse(1.0D)
+      val candidates = scaldingReducers match {
```

Comment:
candidates implies plural here, but from what i can see its just only one candidate possible as a result. so i'd probably update the name